posts/2021-03-24-microprofile40-open-liberty-21003.adoc

```diff
@@ -0,0 +1,442 @@
+
+==== REST metric
+A new `REST.request.unmappedException.total` metric that is backed by a counter metric has been introduced. Similar to the `REST.request` metric, there will be one unique metric for each REST endpoint identified by a class and method label. The new REST metric will count the amount of times the request ends in an unmapped exception. The REST.request metric corresponding to this REST endpoint will not record any values if an unmapped exception has occured.
+
+==== CDI Producer
+The `@Metrics` annotation will no longer support the method target (i.e it can not be annotated on a method). Additionally, it will not support usage with CDI Producers.
+
+==== Try it now
+
+Enable Metrics 3.0 in the `server.xml`, along with any other features you're using.
+
+[source, xml]
+----
+<featureManager>
+    <feature>mpMetrics-3.0</feature>
+</featureManager>
+----
+
+More information:
+
+* link:https://github.com/eclipse/microprofile-metrics/releases/tag/3.0[MicroProfile Metrics 3.0 Release Page (Javadoc & Spec)]
+
+* link:https://download.eclipse.org/microprofile/microprofile-metrics-3.0/microprofile-metrics-spec-3.0.html#release_notes_3_0[Release Notes]
+
+[#health-3.0]
+=== New changes in MicroProfile Health 3.0
+
+MicroProfile Health 3.0 enables you to provide your own health check procedures to be invoked by Open Liberty, to verify the health of your microservices.
+
+MicroProfile Health allows services to report their health, and publish overall health status to a defined endpoint. A service reports "UP" if it is available and reports "DOWN" if it is unavailable. In addition to reporting the state of individual services, MicroProfile Health also aggregates the overall status as "UP" if all the services are "UP". If one of the services is "DOWN" then the overall status is also "DOWN". A service orchestrator can then use the health statuses to make decisions.
+
+MicroProfile Health checks its own health by performing necessary self-checks and then reports its overall status by implementing the API provided by MicroProfile Health. A self-check can be a check on anything that the service needs, such as a dependency, a successful connection to an endpoint, a system property, a database connection, or the availability of required resources. MicroProfile offers checks for both liveness and readiness.
+
+Significant changes in MicroProfile Health 3.0:
+
+* The overall default Readiness status was changed to "DOWN", with an empty response until all the deployed application(s) have been started. A new MicroProfile Config property (`mp.health.default.readiness.empty.response=UP`) is introduced to change the overall default Readiness check status to "UP", during application start up, that do not have any user-defined health checks.
```

Comment:
```suggestion
* The overall default Readiness status was changed to "DOWN", with an empty response until the deployed application(s) have been started. A new MicroProfile Config property (`mp.health.default.readiness.empty.response=UP`) is introduced to change the overall default Readiness check status to "UP", during application start up, that do not have any user-defined health checks.
```

Reply:
Actually I'd be tempted to put the word _all_ in italics rather than remove it

Reply:
Here's how we have it in the docs:

In MicroProfile Health versions 2.0 and later, the overall default readiness status is `DOWN`, with an empty response, until all the deployed applications in Open Liberty are started. In MicroProfile Health versions 3.0 and later, the `mp.health.default.readiness.empty.response=UP` MicroProfile Config property is available. You can specify this property to change the overall default readiness status to `UP` while deployed applications are starting up.